DPGAnalysis/MuonSysAging/plugins/CSCChamberMasker.cc

```diff
@@ -0,0 +1,333 @@
+// -*- C++ -*-
+//
+// Package:    DPGAnalysis/MuonSysAging
+// Class:      CSCChamberMasker
+// 
+/**\class CSCChamberMasker CSCChamberMasker.cc DPGAnalysis/MuonSysAging/plugins/CSCChamberMasker.cc
+
+ Description: Class to mask CSC digis on a chamber by chamber basis
+
+*/
+//
+// Original Author:  Nick J. Amin
+//         Created:  Mon, 27 Feb 2017 15:12:51 GMT
+//
+//
+
+
+// system include files
+#include <map>
+#include <memory>
+#include <string>
+#include <iostream>
+#include <algorithm>
+
+// user include files
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/EDProducer.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Framework/interface/ConsumesCollector.h"
+
+#include "DataFormats/Common/interface/Handle.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/Run.h"
+
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "Geometry/Records/interface/MuonGeometryRecord.h"
+#include "Geometry/CSCGeometry/interface/CSCGeometry.h"
+
+#include "Geometry/CSCGeometry/interface/CSCChamber.h"
+#include "Geometry/CSCGeometry/interface/CSCLayer.h"
+
+#include "DataFormats/CSCDigi/interface/CSCStripDigi.h"
+#include "DataFormats/CSCDigi/interface/CSCStripDigiCollection.h"
+#include "DataFormats/CSCDigi/interface/CSCWireDigi.h"
+#include "DataFormats/CSCDigi/interface/CSCWireDigiCollection.h"
+#include "DataFormats/CSCDigi/interface/CSCCLCTDigi.h"
+#include "DataFormats/CSCDigi/interface/CSCCLCTDigiCollection.h"
+#include "DataFormats/CSCDigi/interface/CSCALCTDigi.h"
+#include "DataFormats/CSCDigi/interface/CSCALCTDigiCollection.h"
+#include "DataFormats/MuonDetId/interface/CSCDetId.h"
+#include "DataFormats/MuonDetId/interface/CSCIndexer.h"
+
+#include "CondFormats/MuonSystemAging/interface/MuonSystemAging.h"
+#include "CondFormats/DataRecord/interface/MuonSystemAgingRcd.h"
+
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "FWCore/Utilities/interface/RandomNumberGenerator.h"
+#include "CLHEP/Random/RandomEngine.h"
+
+
+//
+// class declaration
+//
+
+class CSCChamberMasker : public edm::EDProducer 
```

Comment:
This class needs to be a modern producer (probably an `edm::stream::EDProducer`)